test/cubits/blocklist/blocklist_test.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:hacki/cubits/cubits.dart';
+import 'package:hacki/repositories/preference_repository.dart';
+import 'package:mocktail/mocktail.dart';
+
+class MockPreferenceRepository extends Mock implements PreferenceRepository {}
```

Comment:
I would put this into a single file and directory since this is reusable. e.g. `tests/mocks/repositories/preference_repository.dart`.